include/seqan3/range/container/concatenated_sequences.hpp

```diff
@@ -139,12 +184,22 @@ class concatenated_sequences
      * \brief Static constexpr variables that emulate/encapsulate seqan3::compatible (which doesn't work for types during their definition).
      * \{
      */
+    //!\brief Whether a type satisfies seqan3::compatible with this class's value_type or reference type.
+    //!\hideinitializer
+    // we explicitly check same-ness, because these types may not be fully resolved, yet
+    template <typename t>
+    static constexpr bool is_compatible_value = std::is_same_v<remove_cvref_t<t>, value_type>       ||
+                                                std::is_same_v<remove_cvref_t<t>, reference>        ||
+                                                std::is_same_v<remove_cvref_t<t>, const_reference>  ||
+                                                (dimension_v<t> == dimension_v<value_type> &&
+                                                 std::convertible_to<reference_t<t>, value_type_t<value_type>>);
+    //!\}
```

Comment:
> you mention the seqan3::compatible concept but you do not use it nor are you fully emulating the behaviour?

seqan3::compatible is badly named, badly specified and needs to die by fire.

> Concrete, I am wondering why the last part of the expression should not check the inner most reference type with the inner most value type, but before this was checked (within the seqan3::compatible) concept.

That's exactly what it is doing now, or not?

> I am further wondering where the compatible concept is yet needed, the only other reference was within the translate, but also there it was not used directly but emulated. So either the concept is not valid, or it's definition is wrong?

All of it. It will go away when we also replace our reference_t with iter_reference_t and range_reference_t.

Reply:
> seqan3::compatible is badly named, badly specified and needs to die by fire.

👍 🔥 

> That's exactly what it is doing now, or not?

Ok, I see, the concept is just more generic. In this case we require exactly 2 dimensions?!

> All of it. It will go away when we also replace our reference_t with iter_reference_t and range_reference_t.

Good, because see point 1.